experiment/simulation/js/layout.js

```diff
@@ -0,0 +1,133 @@
+import { simulate, deleteElement } from "./gate.js";
+import { connectGate, unbindEvent, initComparator, refreshWorkingArea } from "./main.js";
+
+"use strict";
+// Wires Colours
+export const wireColours = [
+  "#ff0000",
+  "#00ff00",
+  "#0000ff",
+  "#bf6be3",
+  "#ff00ff",
+  "#00ffff",
+  "#ff8000",
+  "#00ff80",
+  "#80ff00",
+  "#ff0080",
+  "#8080ff",
+  "#c0c0c0",
+];
+
+// Contextmenu
+const menu = document.querySelector(".menu");
+const menuOption = document.querySelector(".menu-option");
+let menuVisible = false;
+
+const toggleMenu = command => {
+  menu.style.display = command === "show" ? "block" : "none";
+  menuVisible = !menuVisible;

```

Comment:
I am a bit uncomfortable with this code. We are setting up the display based on some command but we are flipping visibility each time. Ideally the visibility should be computed from the component state.
What happens here if someone calls toggleMenu("show") twice? The assumption about the toggleMenu code being called alternatively for show and hide breaks and it causes a bug which should not have been there in the first place.
In fact, after reading the code further, I am not sure why you need this menuVisible variable at all.

Reply:
Removed toggle menu and menuvisible, simply using the property menu.style.display to set whether the div is visible or not.